src/common.c

```diff
@@ -926,7 +926,21 @@ header_seek (SF_PRIVATE *psf, sf_count_t position, int whence)
 			if (psf->header.indx + position > psf->header.len)
 			{	/* Need to jump this without caching it. */
 				psf->header.indx = psf->header.end ;
-				psf_fseek (psf, position, SEEK_CUR) ;
+				if (psf->is_pipe)
+				{
+					/* seeking is not supported on pipe input, so we read instead */
+					size_t skip = position ;
```

Comment:
Small issue here (actually an existing bug.) A backwards header seek may have occurred in the buffer, meaning that there are bytes already read an sitting in `header.ptr` between `header.indx` and `header.end`. These bytes need to count against the total in `position` as they have already been read.

So before `psf->header.indx = psf->header.end`, it is needed that `position -= (header.end - header.indx)`.

I found this out when testing if this could help reading ID3v2 headed files from a pipe. Initally we read 12 bytes to try and identify the file header in `guess_file_type()`, so `header.end = 12`. Then `id3_skip()` seeks back to 0 and reads 10 bytes, so `header.indx = 10`. If the ID3 header excluding the parsed 10 bytes is another 10000 bytes long, the correct amount to skip is 10000 - 2 bytes, not 10000.